docs/src/components/Sidebar/Sidebar.tsx

```diff
@@ -46,12 +47,12 @@ class Sidebar extends React.Component<any, any> {
     document.removeEventListener('keydown', this.handleDocumentKeyDown)
   }
 
-  setSearchInput() {
+  private setSearchInput() {
```

Comment:
Class methods are all private in the React paradigm.  I don't think we should be marking every method `private` on in every class.  It is unnecessary, especially for internal only code like docs.

Reply:
> Class methods are all private in the React paradigm

Agreed, but that's the JavaScript paradigm, where anything is publicly accessible anyway. But then comes the TypeScript story / paradigm / however we want to call it.

Since custom React components extend `React.Component`, TypeScript expects component lifecycle classes to actually be public even though they are private from React perspective since user cannot call them):
```
componentDidMount
componentWillUnmount
shouldComponentUpdate
// etc..
```
If we configure TypeScript compiler to make class methods private by default we will need to specify `public` keyword in front of every lifecycle method we define (even though they are private from React perspective).

> I don't think we should be marking every method private on in every class.

Why not? It's the correct thing to do from TypeScript perspective and it will help a lot with detecting dead code (unused methods). I know it's not a valid use-case, but these methods appear as public in case the user gets a reference to a component instance (maybe possible when working with refs?)
![screenshot 2018-11-08 at 17 00 15](https://user-images.githubusercontent.com/5442794/48211503-e35ef800-e379-11e8-89c1-28f527f53cf3.png)

Reply:
this is pending for a while; we can revisit later if necessary